### DISCUSSION TITLE
About detach_reset in the neuron model

### DISCUSSION THREAD

**Discussion by ryime** [fangwei123456/SpikingFlow#505] [Q&A] [answered]
`detach_reset` is a parameter in the neuron model and it is set to `False` by default. In some SNN models, such as [Classify DVS Gesture](https://spikingjelly.readthedocs.io/zh-cn/latest/activation_based_en/classify_dvsg.html), `detach_reset` is set to True.
I want to know how `detach_reset` impacts the performance in SNN training. Do I need to set `detach_reset` to `True` in training large-scale SNN models?
Thank you very much for your help.

**Comment by JackCaster** [comment]
I have the same question---What is the use case of `detach_reset=True`? Perhaps @fangwei123456 can help?

**Comment by fangwei123456** [comment, marked as answer]
> Do I need to set detach_reset to True in training large-scale SNN models?

This is an experimential conclusion from my experience that `detach_reset=True` can improve accuracy.

**Reply by fangwei123456** [reply]
This conclusion is found by https://direct.mit.edu/neco/article/33/4/899/97482/The-Remarkable-Robustness-of-Surrogate-Gradient
